test/gtest/uct/test_uct_ep.cc

```diff
@@ -29,8 +29,13 @@ class test_uct_ep : public uct_test {
 
     void create_sender()
     {
-        m_sender = uct_test::create_entity(0);
-        m_entities.push_back(m_sender);
+        /* Self/Memtype transports must connect to the same ucp_worker */
+        if (has_transport("self") || has_gpu()) {
+            m_sender = m_receiver;
+        } else {
+            m_sender = uct_test::create_entity(0);
+            m_entities.push_back(m_sender);
+        }
```

Comment:
maybe check dev_type for SELF or ACC
like here:
https://github.com/openucx/ucx/blob/d34bcea701db54f257088594febd1a21d4c7aa95/test/gtest/uct/uct_p2p_test.cc#L36